File: internal/content/testdata/bench/go/next-permutation.go
func next_permutation(nums []int) []int {
	nums = append([]int{}, nums...)
	n := len(nums)
	i := n - 2
	for i >= 0 && nums[i] >= nums[i+1] {
		i--
	}
	if i != -1 {
		j := n - 1
		for nums[j] <= nums[i] {
			j--
		}
		nums[i], nums[j] = nums[j], nums[i]
	}
	left, right := i+1, n-1
	for left < right {
		nums[left], nums[right] = nums[right], nums[left]
		left++
		right--
	}
	return nums
}

File: internal/content/testdata/bench/java/next-permutation.java
import java.util.*;

class Solution {
    public long[] next_permutation(long[] nums) {
        long[] arr = Arrays.copyOf(nums, nums.length);
        int n = arr.length;
        int i = n - 2;
        while (i >= 0 && arr[i] >= arr[i + 1]) i--;
        if (i != -1) {
            int j = n - 1;
            while (arr[j] <= arr[i]) j--;
            long tmp = arr[i]; arr[i] = arr[j]; arr[j] = tmp;
        }
        int left = i + 1, right = n - 1;
        while (left < right) {
            long tmp = arr[left]; arr[left] = arr[right]; arr[right] = tmp;
            left++;
            right--;
        }
        return arr;
    }
}
